providers/dynamodb/shedlock-provider-dynamodb/src/main/java/net/javacrumbs/shedlock/provider/dynamodb/DynamodbLockProvider.java

```diff
@@ -0,0 +1,306 @@
+        private final String tableName;
+        private final String hostname;
+        private final Duration minLockTime;
+        private final long concurrentProcessCount;
+
+        DynamodbAccessor(AmazonDynamoDB amazonDynamoDB, String tableName, Duration minLockTime, long concurrentProcessCount) {
+            this.amazonDynamoDB = amazonDynamoDB;
+            this.tableName = tableName;
+            this.hostname = getHostname();
+            this.minLockTime = minLockTime;
+            this.concurrentProcessCount = concurrentProcessCount;
+            try {
+                createTableIfNotExists();
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public boolean insertRecord(LockConfiguration lockConfiguration) {
+
+            DynamoDB dynamoDB = new DynamoDB(amazonDynamoDB);
+            Table table = dynamoDB.getTable(tableName);
+            UpdateItemSpec spec = new UpdateItemSpec()
+                    .withPrimaryKey(NAME, lockConfiguration.getName())
+                    .withUpdateExpression(
+                            String.format("set %s=:%s, %s=:%s, %s=:%s",
+                                    LOCKED_BY, LOCKED_BY, LOCK_UNTIL, LOCK_UNTIL, LOCKED_AT, LOCKED_AT))
+                    .withValueMap(
+                            new ValueMap()
+                                    .withString(":" + LOCKED_BY, lockConfiguration.getName())
+                                    .withLong(":" + LOCK_UNTIL, lockConfiguration.getLockAtMostUntil().toEpochMilli())
+                                    .withLong(":" + LOCKED_AT, Instant.now().toEpochMilli()))
+                    .withReturnValues(ReturnValue.ALL_OLD);
+
+            UpdateItemOutcome updateItemOutcome = table.updateItem(spec);
+            return (updateItemOutcome.getItem() == null);
+        }
+
+        private Map<String, AttributeValue> findLockRecord(LockConfiguration lockConfiguration) {
+            DynamoDB dynamoDB = new DynamoDB(amazonDynamoDB);
+            Table table = dynamoDB.getTable(tableName);
+
+            GetItemSpec spec = new GetItemSpec()
+                    .withPrimaryKey(NAME, lockConfiguration.getName());
+
+            return table.getItemOutcome(spec).getGetItemResult().getItem();
+        }
+
+        private void createTableIfNotExists() throws InterruptedException {
+            if (!isTableExist(tableName)) {
+                createTable(tableName);
+            }
+        }
+
+        private boolean isTableExist(String tableName) {
+
+            // If table has been created already and we are aware,
+            // then skip the expensive operations
+            if (IS_TABLE_EXIST) {
+                return true;
+            }
+
+            // If we are not aware of the table's existance
+            // then find out
+            boolean moreTablesToScan = true;
+            boolean isTableExist = false;
```

Comment:
doesTableExist sounds better

Reply:
This should not be necessary if the createTable method is called only from constructor, so removing this